aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -341,33 +373,43 @@ Tensor& add_out_dense_sparse_cpu(Tensor& r, const Tensor& dense, const SparseTen
     dense.sizes(), " while other has size ", sparse_.sizes(), " (FYI: dense-sparse addition does not currently support broadcasting)");
 
   r.resize_as_(dense);
-  SparseTensor sparse = sparse_.coalesce();
+  if (!is_same_tensor(r, dense)) r.copy_(dense);
+  if (sparse_._nnz() == 0) return r;
+  if (sparse_._values().storage().data() == nullptr) return r;
+  AT_ASSERT(r.has_storage());
 
+  SparseTensor sparse = sparse_.coalesce();
   LongTensor indices = sparse._indices();
   Tensor values = sparse._values();
   int64_t nDim = dense.dim();
   int64_t nDimI = sparse.sparse_dim();
 
-  if (!is_same_tensor(r, dense)) r.copy_(dense);
-  if (sparse._nnz() == 0) return r;
-
   // accessors rely on nnz test
-  if (nDim > nDimI) {
-    auto indices_accessor = indices.accessor<int64_t, 2>();
-    for (int64_t k = 0; k < sparse._nnz(); k++) {
-      Tensor dstBuffer = r;
-      for (int64_t d = 0; d < sparse.sparse_dim(); d++) {
-        dstBuffer = dstBuffer.select(0, indices_accessor[d][k]);
-      }
-      Tensor srcBuffer = values.select(0, k);
-      dstBuffer.add_(srcBuffer, value);
-    }
-  } else {
+  if (nDim == nDimI) {
+    AT_DISPATCH_ALL_TYPES(
+        values.scalar_type(), "add_dense_sparse_scalar", [&] {
+          add_dense_sparse_worker_scalar_cpu<scalar_t>(r, value, sparse, indices, values);
+        });
+  } else if ((r.storage().data() != nullptr) && r.is_contiguous()) {
```

Comment:
What are you trying to check with `r.storage().data() != nullptr` ?

Reply:
this is for one unit test at that time r is zero tensor which has no real data space. 
Since I'm not sure when r.storage().data()==nullptr then it means that's zero tensor, so just have a judge here to avoid